### ISSUE TITLE
Does not match local execution

### ISSUE THREAD

**Issue by adithyabsk** [pre-commit/mirrors-mypy#15]
Accidentally submitted before writing body (WIP)

## Setup
Directory structure
```console
.
├── .git
│   └── ...
├── .mypy_cache
│   └── ...
├── .pre-commit-config.yaml
├── .python-version
├── requirements.txt
└── test
    ├── __init__.py
    └── test.py

```
```python
# test.py
from marshmallow import Schema, fields
import pytz


class TestSchema:
    date = fields.AwareDateTime(default_timezone=pytz.UTC)
```
```
# requirements.txt
mypy==0.770
marshmallow==3.5.1
pytz==2019.3
```
```yaml
# .pre-commit-config.yaml
repos:
-   repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.770
    hooks:
    -   id: mypy

```

Running mypy on the file manually
```console
(some_env) $ python --version
Python 3.7.7
(some_env) $ mypy --version
mypy 0.770
(some_env) $ mypy test/test.py
test/test.py:6: error: Argument "default_timezone" to "AwareDateTime" has incompatible type "_UTCclass"; expected "Optional[timezone]"
Found 1 error in 1 file (checked 1 source file)
```

Running pre-commit on the file
```console
(some_env) $ # Running pre-commit
(some_env) $ pre-commit run -a
mypy.....................................................................Passed
- hook id: mypy
- duration: 0.16s

Success: no issues found in 1 source file
```

## Expectation
The outputs of both should match

## Notes
* I've tried to isolate the issue much to no success
* I've verified that both versions of mypy are the same
* My hunch is that it has something to do with the pre-commit venv but I wasn't able to identify the venv pre-commit uses to debug more (I use pre-commit everywhere and their folder names have guids)

**Comment by adithyabsk** [comment]
@asottile Are there any venv debugging tools that pre-commit has so that I could look into so I can provide a more context for this issue?

**Comment by asottile** [comment]
please read the note in the README: https://github.com/pre-commit/mirrors-mypy#using-mypy-with-pre-commit

**Comment by adithyabsk** [comment]
@asottile I had already read the note in the readme, ~what would I need to do to replicate that run, though?~

Note: for anyone else that comes across this issue in the future, this is how you replicate pre-commit execution from the command line:

```console
$ mypy [FILE_NAME] --ignore-missing-imports --scripts-are-modules --follow-imports skip
```

**Comment by asottile** [comment]
the last section there is the important part:

> Because pre-commit runs mypy from an isolated virtualenv (without your dependencies) you may also find it useful to add the typed dependencies to additional_dependencies so mypy can better perform dynamic analysis:
